CONTRIBUTING.md

```diff
@@ -0,0 +1,104 @@
+
+### Want to contribute the feature yourself?
+
+- **If your feature is a big one,** go ahead and create an issue carefully detailing your idea so that we can coordinate the effort and get you on board.
+- **If your feature is small,** you can go ahead and open up a [Pull Request](https://github.com/Knights-Of-Academia/horace/pulls)
+<br>_Future Proof Note:_ Make some templates for feature requests. One for big changes, one for little ones.
+
+
+## Getting Started With Development<a name = "development"></a>
+
+### Setting Up Your Environment<a name = "setup"></a>
+
+#### Node JS
+
+Horace is based on [NodeJS](https://nodejs.org/en/), which is required to run it. If you haven't already, you'll need ot go ahead and download it.
+
+#### Installing Dependencies
+
+Horace borrows from multiple dependencies. You can go ahead and install these dependencies by running `npm install` in the root directory _after installing NodeJS_.
+
+#### Discord API Usage
+
+If you haven't already realized, Horace is a Discord bot. If you haven't already, go ahead and visit the [Discord Developers Section](https://discordapp.com/developers) in order to create an application for developing bots. This will generate an **API key** for you to use. Connect this API key into the **config.json** file in order to connect the bot authentication ot your instance of the bot.
+<br>_Friendly reminder to keep API keys secret to you. Always remove them from your code, or add them to a .gitignore, before submitting your work._
+
+In order to test Horace, you'll need to invite the bot to your development server. Copy and paste following link, replacing **XXXX** with the **client_id** of your bot. The client ID can be found on the [Developers Section](https://discordapp.com/developers). `https://discordapp.com/oauth2/authorize?client_id=XXXX&scope=bot`
+
+#### The configuration file explained
+
+The file named **config.json.example** contains the environment variables that should be set for the bot to work. _Make sure you have all these variables in your **config.json** with the proper values.
+It requires your Discord API token, COTW channel, Champion Role and Manager Role. _Don't know what these are?_ Check out the [KOA Discord Server](https://discordapp.com/invite/EYX7XGG) to see how these are set up andw hat they're each meant to do.
+
+KOA is heavily involved in [Habitica](http://www.habitica.com/), and your Habitica API Key will be required. You can find [your Habitica API here](https://habitica.com/user/settings/api), just don't forget you'll need to log in.
+In addition to this, it requires your Habitica token, to be able to fetch the COTW information from the Habitica API.
+<br>_Another friendly reminder to keep API keys secret to you. Always remove them from your code, or add them to a .gitignore, before submitting your work._
```

Comment:
I think we should create some templates to streamline this process for people. I'd recommend having bulleted lists, or something similar, such that people can check things off like duplicate issues and whatnot